core/store/models/log_events.go

```diff
@@ -482,8 +489,16 @@ func (parseRunLog20190207withoutIndexes) parseJSON(log eth.Log) (JSON, error) {
 	dataLengthStart := expirationEnd + versionSize + dataLocationSize
 	cborStart := dataLengthStart + dataLengthSize
 
+	if len(log.Data) < dataLengthStart+32 {
+		return JSON{}, errors.New("malformed data")
+	}
+
 	dataLength := whisperv6.BytesToUintBigEndian(data[dataLengthStart : dataLengthStart+32])
 
+	if len(log.Data) < cborStart+int(dataLength) {
+		return JSON{}, errors.New("cbor too short")
+	}
```

Comment:
integer overflow on 32 bit systems: positive -> negative.  Is this checked somewhere?